### PR TITLE
Fees: create new resource

### PR DESCRIPTION
**Pull request by lauren-d**
* Create and adapt fees resource (module)
* Create tests
* Create fixtures
* Update circ_pol JSON Schema and editor
* Create fees when generating notification
* Add default currency in config.py
    
    Co-Authored-by: Aly Badr aly.badr@rero.ch
    Co-Authored-by: Laurent Dubois laurent.dubois@uclouvain.be
    Co-Authored-by: Renaud Michotte renaud.michotte@uclouvain.be
    Co-Authored-by: Benoit Erken benoit.erken@uclouvain.be

How to test:
- run bootstrap
- run setup
- you can display the fee at url : /api/fees/1

### CODE REVIEW

File: rero_ils/modules/fees/api.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+                patron_type_pid,
+                holding_circulation_category_pid
+            )
+            data['amount'] = cipo.get('overdue_amount')
+            data['currency'] = current_app.config\
+                .get('RERO_ILS_DEFAULT_CURRENCY')
+            data['status'] = 'open'
+            record = Fee.create(
+                data, dbcommit=True, reindex=True, delete_pid=True)
+            return record
```
Comment (rerowep) [inline rero_ils/modules/fees/api.py]: What do we return if we do not have a notification_type == `overdue`?

File: tests/api/test_fees.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Tests REST API notifications."""
+
```
[the remark this answers is not in the document]
Comment (lauren-d) [reply tests/api/test_fees.py]: Done

File: rero_ils/modules/fees/api.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""API for manipulating Fees."""
+
```
[the remark this answers is not in the document]
Comment (lauren-d) [reply rero_ils/modules/fees/api.py]: Done

File: rero_ils/modules/fees/utils.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
```
[the remark this answers is not in the document]
Comment (lauren-d) [reply rero_ils/modules/fees/utils.py]: Done

File: rero_ils/modules/fees/api.py [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+    fetcher = fee_id_fetcher
+    provider = feeProvider
+    model_cls = FeeMetadata
+
+    def create_fee_from_notification(notification):
+        """Create a new fee."""
+        if notification.get('notification_type') == 'overdue':
+            data = {}
+            schemas = current_app.config.get('RECORDS_JSON_SCHEMA')
+            data_schema = {
```
[the remark this answers is not in the document]
Comment (lauren-d) [reply rero_ils/modules/fees/api.py]: Code copy from another class

File: rero_ils/modules/fees/api.py [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+class Fee(IlsRecord):
+    """Fees class."""
+
+    minter = fee_id_minter
+    fetcher = fee_id_fetcher
+    provider = feeProvider
+    model_cls = FeeMetadata
+
+    def create_fee_from_notification(notification):
```
[the remark this answers is not in the document]
Comment (lauren-d) [reply rero_ils/modules/fees/api.py]: Done

File: rero_ils/modules/fees/api.py (hunk 1 above)
Comment (lauren-d) [reply rero_ils/modules/fees/api.py]: Done

File: rero_ils/modules/fees/api.py [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+            cipo = CircPolicy.provide_circ_policy(
+                library_pid,
+                patron_type_pid,
+                holding_circulation_category_pid
+            )
+            data['amount'] = cipo.get('overdue_amount')
+            data['currency'] = current_app.config\
+                .get('RERO_ILS_DEFAULT_CURRENCY')
+            data['status'] = 'open'
+            record = Fee.create(
```
[the remark this answers is not in the document]
Comment (lauren-d) [reply rero_ils/modules/fees/api.py]: Done

File: rero_ils/modules/fees/api_views.py [hunk 8]
```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
```
Comment (rerowep) [inline rero_ils/modules/fees/api_views.py]: This file is normally called `views.py`.

File: rero_ils/modules/loans/api.py [hunk 9]
```diff
@@ -291,3 +291,19 @@ def get_overdue_loans():
         if now > due_date + timedelta(days=days_after):
             overdue_loans.append(loan)
     return overdue_loans
+
+
+def get_item_on_loan_loans():
+    """Return item on loan."""
+    from .utils import get_circ_policy
```
Comment (rerowep) [inline rero_ils/modules/loans/api.py]: Could we put this on top of file?

File: rero_ils/modules/fees/api_views.py (hunk 8 above)
Comment (lauren-d) [reply rero_ils/modules/fees/api_views.py]: Done

File: rero_ils/modules/loans/api.py (hunk 9 above)
Comment (lauren-d) [reply rero_ils/modules/loans/api.py]: Done

File: data/circulation_policies.json [hunk 10]
```diff
@@ -11,6 +11,7 @@
     "allow_requests": true,
     "number_of_days_before_due_date": 5,
     "number_of_days_after_due_date": 5,
+    "overdue_amount": 2,
```
[the remark this answers is not in the document]
Comment (lauren-d) [reply data/circulation_policies.json]: Fixed

File: rero_ils/modules/circ_policies/jsonschemas/circ_policies/circ_policy-v0.0.1.json [hunk 11]
```diff
@@ -79,6 +79,12 @@
       "default": 5,
       "minimum": 1
     },
+    "overdue_amount": {
+      "title": "Overdue amount",
+      "description": "Overdue amount.",
+      "type": "number",
+      "default": 2
```
[the remark this answers is not in the document]
Comment (lauren-d) [reply rero_ils/modules/circ_policies/jsonschemas/circ_policies/circ_policy-v0.0.1.json]: Fixed

File: tests/data/data.json [hunk 12]
```diff
@@ -700,6 +700,7 @@
     "checkout_duration": 30,
     "number_of_days_before_due_date": 5,
     "number_of_days_after_due_date": 5,
+    "overdue_amount": 2,
```
[the remark this answers is not in the document]
Comment (lauren-d) [reply tests/data/data.json]: Fixed

**Pull request merged**